src/main.rs

```diff
@@ -47,26 +47,40 @@ fn main() -> Result<(), JaktError> {
                     } else {
                         String::from("runtime")
                     };
-                    let default_cxx_compiler_path = PathBuf::from("clang++");
-                    let clang_output = Command::new(
-                        arguments
-                            .cxx_compiler_path
-                            .as_ref()
-                            .unwrap_or(&default_cxx_compiler_path),
-                    )
-                    .args([
-                        "-fcolor-diagnostics",
-                        "-std=c++20",
-                        // These warnings if enabled create loads of unnecessary noise:
-                        "-Wno-unqualified-std-cast-call",
-                        "-Wno-user-defined-literals",
-                        "-I",
-                        &runtime_path,
-                        &input_cpp,
-                        "-o",
-                        &output_executable,
-                    ])
-                    .output()?;
+                    let cxx_compiler = if let Some(ref path) = arguments.cxx_compiler_path {
+                        path.to_owned()
+                    } else {
+                        PathBuf::from(env!("CXX", "No supported C++ compiler was found!"))
+                    };
+
+                    let is_clang = cxx_compiler.to_string_lossy().contains("clang");
+
+                    let color_diags = if is_clang {
+                        "-fcolor-diagnostics"
+                    } else {
+                        "-fdiagnostics-color"
+                    };
+
+                    let def_literals = if is_clang {
+                        "-Wno-user-defined-literals"
+                    } else {
+                        "-Wno-literal-suffix"
+                    };
+
+                    let clang_output = Command::new(cxx_compiler)
```

Comment:
Also this variable should now be `cxx_compiler_output` since it's not just clang.